theme/assets/js/templates/store/item.js

```diff
@@ -495,6 +497,103 @@ document.addEventListener('alpine:init', () => {
         },
     });
 
+    const createAfterpayBadgeData = () => ({
+        payments: null,
+        paymentAmount: null,
+        isLoadingAfterpayBadge: false,
+        selector: '.afterpay-placement',
+        /**
+         * Initialize Square Web Payments SDK
+         *
+         * @param {String} applicationId the application id
+         * @param {String} locationId the location id
+         * @return {Object|null}
+        */
+        initWebPaymentsSDK(applicationId, locationId) {
+            if (!this.webPaymentsSDKLoaded()) {
+                throw new Error('Square Web Payments SDK is not loaded');
+            }
+
+            const Square = window.Square;
+            return Square.payments(applicationId, locationId);
+        },
+
+        /**
+         * Whether the Web Payments SDK has already been loaded
+         * @return {Boolean}
+         */
+        webPaymentsSDKLoaded() {
+            return !!window.Square;
+        },
+        async init() {
+            // Initialize the payments SDK
+            const applicationId = Alpine.store('global').applicationId;
+            const locationId = Alpine.store('product').locationId;
+            const storeLocale = Alpine.store('global').locale;
+            const payments = this.initWebPaymentsSDK(applicationId, locationId);
+            payments.setLocale(storeLocale);
+            this.payments = payments;
+
+            this.paymentAmount = this.product.price.regular_low.amount;
+
+            // Load afterpay messaging
+            await this.attachAfterpayMessaging();
+            this.$watch('$store.product.selectedVariationPrice', () => this.reloadAfterpayMessaging());
+        },
+        reloadAfterpayMessaging() {
+            this.cleanupAfterpayMessaging();
+            this.paymentAmount = Alpine.store('product').selectedVariationPrice;
+            this.attachAfterpayMessaging();
+        },
+        async attachAfterpayMessaging() {
+            this.isLoadingAfterpayBadge = true;
+            try {
+                const paymentRequest = this.buildPaymentRequest(this.paymentAmount);
+                const afterpay = await this.payments.afterpayClearpay(paymentRequest);
+                this.cleanupAfterpayMessaging();
+                await afterpay.attachMessaging(this.selector, Constants.AFTERPAY_MESSAGING_OPTIONS);
+            } catch (err) {
+                throw new Error('Failed to attach afterpay messaging');
+            } finally {
+                this.isLoadingAfterpayBadge = false;
+            }
+        },
+
+        /**
+         * Remove all afterpay messages that have been attached to this badge
+         */
+        cleanupAfterpayMessaging() {
+            // Container may not exist if there was previously an error
+            if (this.$refs.container) {
+                const afterpayMessages = this.$refs.container.querySelectorAll('afterpay-placement');
```

Comment:
Should we change `selector` to `selectorName: 'afterpay-placement'` so it can be used here too?